middleware/index.ts

```diff
@@ -1,25 +1,66 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
 /* eslint-disable @typescript-eslint/no-explicit-any */
+const path = require('path');
 import {Logger, LoggerFormat, LoggerLevel} from '@salesforce/core/lib/logger';
 import {CloudEvent,Headers as CEHeaders,Receiver} from 'cloudevents';
 const {Message} = require('@projectriff/message');
-const http = require('http');
-const https = require('https');
 import {applySfFnMiddleware} from './lib/sfMiddleware';
-import {
-    FunctionInvocationRequest,
-    saveFnInvocation,
-    saveFnInvocationError
-} from './lib/FunctionInvocationRequest';
-import {
-    ASYNC_CE_TYPE,
-    ASYNC_FULFILL_HEADER,
-    FN_INVOCATION,
-    X_FORWARDED_HOST,
-    X_FORWARDED_PROTO
-} from './lib/constants';
 import loadUserFunction from './userFnLoader';
+import { Context, InvocationEvent } from '@salesforce/salesforce-sdk';
+
+const FUNCTION_ERROR_CODE = '500';
+const INTERNAL_SERVER_ERROR_CODE = '503';
+const CURRENT_FILENAME: string = __filename;
+
+export class ExtraInfo {
+    constructor(
+        public readonly requestId: string,  // incoming x-request-id
+        public readonly source: string,     // incoming ce.source
+        public readonly execTimeMs: number, // function invocation time
+        public stack = ''          // error stack, if applicable
+        ) {
+        this.stack = encodeURI(this.trim(stack));
```

Comment:
I think the URI-encoding should happen for the whole structure as it's added to the header, rather than here.